spring-cloud-stream-binders/spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -435,51 +387,29 @@ public void afterPropertiesSet() throws Exception {
 	}
 
 	@Override
-	public void bindConsumer(final String name, MessageChannel moduleInputChannel, Properties properties) {
+	public void doBindConsumer(String name, String group, MessageChannel inputChannel, Properties properties) {
+		String baseQueueName = BinderUtils.groupedName(name, group);
 		if (this.logger.isInfoEnabled()) {
-			this.logger.info("declaring queue for inbound: " + name);
-		}
-		if (name.startsWith(P2P_NAMED_CHANNEL_TYPE_PREFIX)) {
-			validateConsumerProperties(name, properties, SUPPORTED_NAMED_CONSUMER_PROPERTIES);
-		}
-		else {
-			validateConsumerProperties(name, properties, SUPPORTED_CONSUMER_PROPERTIES);
+			this.logger.info("declaring queue for inbound: " + baseQueueName + ", bound to: " + name);
 		}
 		RabbitPropertiesAccessor accessor = new RabbitPropertiesAccessor(properties);
-		String queueName = applyPrefix(accessor.getPrefix(this.defaultPrefix), name);
-		TopicExchange exchange = new TopicExchange(queueName);
-		declareExchange(queueName, exchange);
+		validateConsumerProperties(baseQueueName, properties, SUPPORTED_CONSUMER_PROPERTIES);
+		String prefix = accessor.getPrefix(this.defaultPrefix);
+		String exchangeName = applyPrefix(prefix, name);
+		TopicExchange exchange = new TopicExchange(exchangeName);
+		declareExchange(exchangeName, exchange);
+
+		String queueName = applyPrefix(prefix, baseQueueName);
 		int partitionIndex = accessor.getPartitionIndex();
-		String dlqNameRoot = name;
+		String dlqNameRoot = baseQueueName;
 		if (partitionIndex >= 0) {
 			String partitionSuffix = "-" + partitionIndex;
 			queueName += partitionSuffix;
 			dlqNameRoot += partitionSuffix;
```

Comment:
https://github.com/garyrussell/spring-cloud-stream/commit/4e8d56bff9b828198d8921c1035141d82a28050a

I didn't push directly to your branch in case you have some local WIP.